example-datastream-job/src/main/java/org/apache/flink/connector/prometheus/examples/DataStreamJob.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.flink.connector.prometheus.sink.PrometheusRequestSigner;
+import org.apache.flink.connector.prometheus.sink.PrometheusSink;
+import org.apache.flink.connector.prometheus.sink.PrometheusTimeSeries;
+import org.apache.flink.connector.prometheus.sink.PrometheusTimeSeriesLabelsAndMetricNameKeySelector;
+import org.apache.flink.connector.prometheus.sink.aws.AmazonManagedPrometheusWriteRequestSigner;
+import org.apache.flink.connector.prometheus.sink.errorhandling.OnErrorBehavior;
+import org.apache.flink.connector.prometheus.sink.errorhandling.SinkWriterErrorHandlingBehaviorConfiguration;
+import org.apache.flink.connector.prometheus.sink.http.RetryConfiguration;
+import org.apache.flink.connector.prometheus.sink.prometheus.Types;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+/** Test application testing the Prometheus sink connector. */
+public class DataStreamJob {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataStreamJob.class);
+
+    public static void main(String[] args) throws Exception {
+        StreamExecutionEnvironment env;
+
+        // Conditionally return a local execution environment with
+        if (args.length > 0 && Arrays.stream(args).anyMatch("--webUI"::equalsIgnoreCase)) {
+            Configuration conf = new Configuration();
+            conf.set(
+                    ConfigOptions.key("rest.flamegraph.enabled").booleanType().noDefaultValue(),
+                    true);
+            env = StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(conf);
+        } else {
+            env = StreamExecutionEnvironment.getExecutionEnvironment();
+        }
+
+        env.setParallelism(2);
+
+        ParameterTool applicationParameters = ParameterTool.fromArgs(args);
+
+        // Prometheus remote-write URL
+        String prometheusRemoteWriteUrl = applicationParameters.get("prometheusRemoteWriteUrl");
+        LOGGER.info("Prometheus URL:{}", prometheusRemoteWriteUrl);
+
+        // Optionally configure Amazon Managed Prometheus request signer
+        PrometheusRequestSigner requestSigner = null;
+        String ampAWSRegion = applicationParameters.get("awsRegion");
+        if (ampAWSRegion != null) {
+            requestSigner =
+                    new AmazonManagedPrometheusWriteRequestSigner(
+                            prometheusRemoteWriteUrl, ampAWSRegion);
+            LOGGER.info(
+                    "Enable Amazon Managed Prometheus request-signer, region: {}", ampAWSRegion);
+        }
+
+        // Configure data generator
+        int generatorMinSamplesPerTimeSeries = 1;
+        int generatorMaxSamplesPerTimeSeries = 10;
+        int generatorNumberOfDummyInstances = 5;
```

Comment:
Hmm, this is hard to understand out of context. Can we add an overall description at the top of this class, or rename this to "NumberOfKeys/NumberOfUniqueMetrics" or something?

Reply:
Reworked the random data generator to generate a configurable random number of metrics (`M0000`, `M00001`...) for a configurable random number of sources (`SourceID` = `S0000000000`, `S0000000001`...).
Removed confusing references to "instances", "CPU" and "Memory"